### ISSUE TITLE
Intercom::Request#parse_body fails with Encoding::InvalidByteSequenceError

### ISSUE THREAD

**Issue by thesmart** [intercom/intercom-ruby#251]
I suppose one of our users has unicode data in the data we sync with Intercom.  Intercom::Request#parse_body is raising the following error:
```
Encoding::InvalidByteSequenceError: "\xC3" on US-ASCII
  from json/common.rb:155:in `encode'
  from json/common.rb:155:in `initialize'
  from json/common.rb:155:in `new'
  from json/common.rb:155:in `parse'
  from intercom/request.rb:90:in `parse_body'
  from intercom/request.rb:70:in `block in execute'
  from net/http.rb:853:in `start'
  from intercom/request.rb:65:in `execute'
  from intercom/client.rb:94:in `execute_request'
  from intercom/client.rb:76:in `get'
  from intercom/api_operations/find.rb:10:in `find'
```

Perhaps the client and intercom server aren't agreeing on UTF-8?

## Version info
  - intercom-ruby version: 3.3.0
  - Ruby version: 2.1.5

## Expected behavior
Not fail.

## Actual behavior
Fail.

## Steps to reproduce
Not sure, this occurred in production.  Here's the code on our side:
```ruby
      user_hash = intercom_user_hash(user: user, user_id: user_id)

      begin
        i_user ||= intercom.users.find({ user_id: user_hash[:user_id] })
        user_hash.each do |k,v|
          i_user.send(:"#{k}=", v)
        end
        i_user = intercom.users.save(i_user)
      rescue ::Intercom::ResourceNotFound
        i_user = intercom.users.create(user_hash)
      end
```

## Logs
Stack trace above.

**Comment by thesmart** [comment]
This is a known Ruby issue:
https://bugs.ruby-lang.org/issues/2567

You guys need to force the encoding to match your servers'.  ```response.body.force_encoding(Encoding::UTF_8)```

**Comment by choran** [comment]
Hi, 
We updated this to force encode it as you indicated. 
This should now be available in Gem "3.5.1".
Can you take a look at this and check if it fixes your issue?
If not feel free to re-open this issue and let us know what you are seeing and we will take a look
